File: 0997-find-the-town-judge/0997-find-the-town-judge.java
class Solution {
    public int findJudge(int n, int[][] trust) {
        if(trust.length == 0) return n == 1 ? 1 : -1;
        int[] balance = new int[n + 1];
        for(int[] t : trust) {
            --balance[t[0]];
            ++balance[t[1]];
        }
        for(int i = 0; i <= n; ++i)
            if(balance[i] == n - 1) 
                return i;
        return -1;
    }
}

File: 0997-find-the-town-judge/0997-find-the-town-judge.py
class Solution:
    def findJudge(self, n: int, trust: List[List[int]]) -> int:
        if len(trust) == 0:
            return n if n is 1 else -1
        balance = [0] * (n + 1)
        for t in trust:
            balance[t[0]] -= 1
            balance[t[1]] += 1
        for i in range (0, n + 1):
            if balance[i] == n - 1:
                return i
        return -1